### ISSUE TITLE
Deadline for deliverable HTML export and agreement on final HTML export

### ISSUE THREAD

**Issue by atomrab** [atomrab/chersark#27]
We should set a deadline for the initial HTML export of the DB, which may or may not come before I've finished finalizing all the records (but shouldn't come until the other issues have been resolved). If the deadline needs to be sooner so that the invoicing can be completed, I would like to set a secondary export date by which time I will have completed all the updates, edits, and revisions to the database so that we can have an archival copy.

It is possible that I will be able to make an arrangement with TACC to add the publication cleaned versions of photos and drawings to the iRODS installation where the image resources are currently stored. If I can do that, does the ARK page send an automatic query to look for files of a certain format, or were links to external resources hard-coded at a particular moment? If there is a way to update those TACC links to pull in the new material, there may also need to be a third moment of export in a few months when I've settled on long-term arrangements with TACC.

**Comment by MikeJJohnson** [comment]
Do you have an estimate for when you are completing the edits and revisions? We already have an export run last month which has the old data, so I don't foresee any hiccups for the future.

The url's for the images are stored in the database - if there is a bulk change that can be made (eg tif to jpg extensions) that is no problem but there isn't anything dynamic about fetching of the image.

I think that any invoicing deadlines will be from the university side, I know that they were expecting the invoice to be dated in August. I can still put an august date on the invoice even if it is issued later this month when you are happy.

**Comment by atomrab** [comment]
> Do you have an estimate for when you are completing the edits and revisions? We already have an export run last month which has the old data, so I don't foresee any hiccups for the future.

I am hoping to have this mostly done by the end of this month, but I seem to hope a lot of things that don't come true. Certainly not any later than the end of December. Is it just pressing a button and waiting on your side, or is it more complicated than that?

Also, are the ARK shoulders and pass-through suffixes added on export, or do we get a permalink field on each page with the n2t URL and the relative suffix before export? 

> The url's for the images are stored in the database - if there is a bulk change that can be made (eg tif to jpg extensions) that is no problem but there isn't anything dynamic about fetching of the image.

It's less about a bulk change than about adding a bunch of new URLs for new assets. Was there a script that generated the URLs the first time around, with some sort of a call to the iRODS collection? If so, we could add to iRODS and re-run the script, but it would have to be before the final export. If not, I guess it would be manual page editing for me to make the connection with the new resources (i.e. publication photos, all of the image resources for CH01 and CH02, etc).

> I think that any invoicing deadlines will be from the university side, I know that they were expecting the invoice to be dated in August. I can still put an august date on the invoice even if it is issued later this month when you are happy.

They're not super strict about enforcing contractual end-dates, and we already have a no-cost extension for the grant this is coming from until the end of September. But we don't want to wait too long. I think running another export in mid-September and putting in the invoice right after that is the best bet -- gives us enough time to deal with any problems with either the export or the invoice (make sure the L - P accountant sends it to Classics and not the Library this time), but also gives me another week to see if anything still needs fixing.

**Comment by atomrab** [comment]
Let's not forget about the following question -- let me know if I should bump it to its own issue.

> Also, are the ARK shoulders and pass-through suffixes added on export, or do we get a permalink field on each page with the n2t URL and the relative suffix before export?

**Comment by atomrab** [comment]
Assuming that we're on the same page about running a push-button export some time in November, and since the invoice has been received and processed, I'm closing this issue.
